### ISSUE TITLE
Stack overflow in isExportsOrModuleExportsOrAlias 

### ISSUE THREAD

**Issue by amcasey** [microsoft/TypeScript#31080]
Originally reported on Developer Community: https://developercommunity.visualstudio.com/content/problem/529989/aspnet-在html-文件-js-智能提示失效.html

Stack cycle is 
```
at isExportsOrModuleExportsOrAlias (C:\Program Files (x86)\Microsoft SDKs\TypeScript\3.3\tsserver.js:29774:38)
at isExportsOrModuleExportsOrAliasOrAssignment (C:\Program Files (x86)\Microsoft SDKs\TypeScript\3.3\tsserver.js:29783:16)
at isNameOfExportsOrModuleExportsAliasDeclaration (C:\Program Files (x86)\Microsoft SDKs\TypeScript\3.3\tsserver.js:29780:54)
at isExportsOrModuleExportsOrAlias (C:\Program Files (x86)\Microsoft SDKs\TypeScript\3.3\tsserver.js:29774:38)
```

TS version is 3.3.3.

I looked at the code and didn't see an obvious cycle (unless a node is somehow its own initializer), so I'm guessing it was just a deep stack of nested initializers.

I didn't see a recent change in this area or a but with these stack frames.

**Comment by rbuckton** [comment]
It could be that this is where the stack terminates, but isn't itself cyclical. Is there more context? I cannot seem to see where you found the stack trace.

**Comment by rbuckton** [comment]
It looks like this has already been fixed.

**Comment by rbuckton** [comment]
This was fixed by #31436
